test/hardhat/blocklock.test.ts

```diff
@@ -161,13 +161,26 @@ describe("Blocklock integration tests", () => {
       wallet,
     );
     expect(await mockBlocklockReceiverInstance.plainTextValue()).to.equal(BigInt(0));
+
     const blockHeight = BigInt((await ethers.provider.getBlockNumber()) + 10);
     console.log("block height", blockHeight);
+
+    // condition bytes
+    const types = ["string", "uint256"]; // "B" is a string, and blockHeight is a uint256
+    const values = ["B", blockHeight];
```

Comment:
will all conditions be a single char? If not, have we thought about extension attacks against this? I'm sure we have, but thought i'd flag anyway

Reply:
That's right. The agent can handle the checks off-chain and the js lib can enforce the condition encoding in the correct formats given an input, e.g., block number. We've offloaded the condition parsing and checking off-chain for gas and to avoid spending gas on setting supported condition prefixes on-chain.

Reply:
We currently a assume single char, but we can change the encoding (or chain it, e.g., a new letter C, which wraps a new encoding)

Reply:
Sounds good.

Reply:
in retrospect this gives us all UTF8 chars which is quite a lot of different conditions